parlai/tasks/jericho_world/agents.py

```diff
@@ -202,6 +210,62 @@ def _clean_example(self, game_step: Dict[str, Any]) -> Dict[str, Any]:
             example['next_state'] = extract_state_data(game_step['next_state'])
         return example
 
+    def extract_knowledge_graph_str(
+        self, state: Dict[str, Any], include_inv_objs: bool = False
+    ) -> str:
+        """
+        Generates the string representing knowledge graph.
+
+        It may modifiy the knowledge graph, for example when --prune-knowledge-graph is true.
+        """
+
+        def has_word_overlap(main_text, content_text_tokens):
+            if not main_text or not content_text_tokens:
+                return False
+
+            for word in main_text.lower().split():
+                if not word or len(word) < 3 or word in consts.GRAPH_VERT_SKIP_TOKEN:
+                    continue
+                if word in content_text_tokens:
+                    return True
+            return False
+
+        def keep_edge(edge, text_tokens):
```

Comment:
I'm not sure I follow what your `keep_edge` heuristic is here - so we keep all 'you in X' and 'you have X', but otherwise we keep edges where the subject and object are in the text? Probably could use a quick one line desc to this effect.

```
'''
Returns true if all the components of the edge are in the given text, or if it is the player location or unpruned inventory
'''
```

Sure this documentation isn't required, but it makes it easier to grok what's happening.

Reply:
Explained it in details in the method docstring.